frontend/src/main/scala/akkaviz/frontend/components/FsmGraph.scala

```diff
@@ -0,0 +1,59 @@
+package akkaviz.frontend.components
+
+import akkaviz.frontend.vis
+import akkaviz.frontend.vis.{NetworkData, NetworkOptions}
+import org.scalajs.dom.html.Element
+import org.scalajs.dom.{Element => domElement, _}
+
+import scala.scalajs.js
+import scala.scalajs.js.JSConverters._
+
+class FsmGraph(parent: Element) {
+
+  private[this] val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] val data = NetworkData(networkNodes, networkEdges)
+  private[this] val network = new vis.Network(parent, data, fsmNetworkOptions)
+
+  network.on("stabilizationProgress", () => network.fit())
+  network.on("stabilized", () => network.fit())
+
+  def displayFsm(transitions: Set[(String, String)]): Unit = {
+    if (transitions.isEmpty) {
+      parent.style.display = "none"
+    } else {
+      parent.style.display = "box"
+
+      val allStates = transitions.flatMap { case (from, to) => Set(from, to) }
+      val nodes = allStates.toJSArray.map(state => vis.Node(state, simplifyStateName(state), state))
+      val edges = transitions.toJSArray.map {
+        case (from, to) => vis.Edge(s"${from}->${to}", from, to)
+      }
+      networkNodes.clear()
+      networkNodes.add(nodes)
+      networkEdges.clear()
+      networkEdges.add(edges)
+    }
+  }
+
+  @inline
+  private[this] def simplifyStateName(state: String): String = {
+    state.split('.').last.stripSuffix("$")
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of last on collections.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2167023948/issues/source?bid=3121282&fileBranchId=3161575#l41)